From 2c82208c37d2e1c2a2b795e3db0a3c9a145695ca Mon Sep 17 00:00:00 2001
From: Nacho Garcia <ngarcia@fluendo.com>
Date: Mon, 1 Oct 2018 15:48:44 +0200
Subject: [PATCH] Explicitly added SSE support

Returning SSE vectors without explicitly enabling
SSE support causes an error on new mingw versions
This patch explicitly enables SSE if available,
setting flags up to SSE4.1 version
---
 configure.ac                  | 17 +++++++++++++++++
 gst/audioresample/Makefile.am |  5 ++++-
 2 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 2d24f3e..3b4ad26 100644
--- a/configure.ac
+++ b/configure.ac
@@ -189,6 +189,23 @@ AC_COMPILE_IFELSE(
 CPPFLAGS="$ac_cppflags_save"
 AM_CONDITIONAL(HAVE_LIBXML_HTML, test "x$HAVE_LIBXML_HTML" = "xyes")
 
+dnl check for -m* compiler flags too
+SSE_CFLAGS="-msse"
+SSE2_CFLAGS="-msse2"
+SSE41_CFLAGS="-msse4.1"
+
+AS_COMPILER_FLAG([$SSE_CFLAGS], [HAVE_SSE=1], [HAVE_SSE=0])
+AS_COMPILER_FLAG([$SSE2_CFLAGS], [HAVE_SSE2=1], [HAVE_SSE2=0])
+AS_COMPILER_FLAG([$SSE41_CFLAGS], [HAVE_SSE41=1], [HAVE_SSE41=0])
+
+AC_DEFINE_UNQUOTED(HAVE_SSE, [$HAVE_SSE], [SSE support is enabled])
+AC_DEFINE_UNQUOTED(HAVE_SSE2, [$HAVE_SSE2], [SSE2 support is enabled])
+AC_DEFINE_UNQUOTED(HAVE_SSE41, [$HAVE_SSE41], [SSE4.1 support is enabled])
+
+AC_SUBST(SSE_CFLAGS)
+AC_SUBST(SSE2_CFLAGS)
+AC_SUBST(SSE41_CFLAGS)
+
 dnl used in gst/tcp
 AC_CHECK_HEADERS([sys/socket.h],
   HAVE_SYS_SOCKET_H="yes", HAVE_SYS_SOCKET_H="no")
diff --git a/gst/audioresample/Makefile.am b/gst/audioresample/Makefile.am
index dc02daf..9150972 100644
--- a/gst/audioresample/Makefile.am
+++ b/gst/audioresample/Makefile.am
@@ -17,7 +17,10 @@ libgstaudioresample_la_CFLAGS = \
 	$(GST_PLUGINS_BASE_CFLAGS) \
 	$(GST_BASE_CFLAGS) \
 	$(GST_CFLAGS) \
-	$(ORC_CFLAGS)
+	$(ORC_CFLAGS) \
+	$(SSE_CFLAGS) \
+	$(SSE2_CFLAGS) \
+	$(SSE41_CFLAGS)
 
 libgstaudioresample_la_LIBADD = \
 	$(GST_BASE_LIBS) \
-- 
2.7.4

